streams/src/test/java/org/apache/kafka/streams/processor/internals/ProcessorStateManagerTest.java

```diff
@@ -441,6 +441,123 @@ public void shouldNotChangeOffsetsIfAckedOffsetsIsNull() throws IOException {
         assertThat(read, equalTo(offsets));
     }
 
+    @Test
+    public void shouldIgnoreIrrlevantLoadedCheckpoints() throws IOException {
```

Comment:
Added a bunch of new tests to cover both the bug itself and also previously untested code paths in the ProcessorStateManager.

Reply:
Note: this particular test shows that we will actually repair all the corrupted checkpoint files that buggy Streams versions wrote.